woocommerce/woocommerce-ios
# Order Creation: Update customer section

Part of https://github.com/woocommerce/woocommerce-ios/issues/5412.
Closes https://github.com/woocommerce/woocommerce-ios/issues/5808.

## Description

This PR updates layout for customer section and fixes some inconsistencies and bugs.

## Changes

Most work here cherry-picked from closed https://github.com/woocommerce/woocommerce-ios/pull/5818.

- Updates `OrderCustomerSection` and `OrderStatusSection` to use safeAreaInsets from Environment.
- Updates `OrderCustomerSection` layout.
- Separates `OrderCustomerSectionContent` to hold redrawn content of `OrderCustomerSection`, fixing earlier navigation and state glitches.
- Updates `EditOrderAddressForm` to disable:
  - autocorrection on all fields
  - autocapitalization on email field

### Known limitations

Separate shipping/billing address editing is not available here.

## Testing

1. Build and run the app in debug mode (to ensure the feature flag is enabled).
2. Make sure Order Creation is enabled under Settings > Experimental Features.
3. Go to the Orders tab and tap the + button. Tap "Create order" in the bottom sheet.
4. Tap "+ Add customer" button to display address form.
5. Enter some data.
6. Tap "Done" to save and close address form.
7. Confirm that customer section is updated with new data.
8. Tap "Edit" in customer section.
9. Confirm that address form. shows correct data.
10. Clear all fields and tap "Done".
11. Confirm that empty state for both addresses shows "No address specified.".

## Screenshots

Empty state|Entered Addresses|Empty state
--|--|--
![Simulator Screen Shot](https://user-images.githubusercontent.com/3132438/148964937-e4e47631-ab45-4430-a7d1-cfecd99d58f9.png)|![Simulator Screen Shot](https://user-images.githubusercontent.com/3132438/148964948-fab8e5fd-1fdb-4b1b-bfae-6d05681d82ea.png)|![Simulator Screen Shot](https://user-images.githubusercontent.com/3132438/148964966-8a3f5ff4-adbc-4a3d-b9db-08862489fe10.png)


---
